src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -70,6 +80,38 @@ internal void PopulateAutoCompleteCandidates()
             }
         }
 
+        /// <summary>
+        /// Filters the matching node search elements based on user input in the search field. 
+        /// </summary>
+        internal void SearchAutoCompleteCandidates(string input)
+        {
+            if (PortViewModel == null) return;
+
+            FilteredResults = searchElementsCache.Where(e =>
+                                QuerySearchElements(e, input)).Select(e =>
```

Comment:
Can we not use the same search algorithm as the one used for in-canvas search?

Reply:
For the incanvas search we use the search command on the model and search for all the nodes in the library. It doesn't search a specific list of nodes.

Reply:
Yes, I got that but can we not use the same algorithm/methods, refactor some code, etc?

Reply:
@aparajit-pratap Can you be more specific? We are not using the same algorithm here right?

Reply:
Yes, Aparajit is asking if we could use the same algorithm to search. https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Search/SearchDictionary.cs#L327

We can use it with some changes but since the list of search elements won't be that long and we are checking just the name, I kept it simple without using the weights and time. What do you guys think?

Reply:
I think this looks good for now.

Reply:
I think we should keep the search algorithm consistent irrespective of how many search elements there might be. I think one of @Amoursol's points of emphasis was that search is key for the node autocomplete feature. We should at least create an improvement task for this to address in the near future.

Reply:
Indeed @aparajit-pratap - now that I'm back we will spin up a conversation around Search :) For Node Autocomplete to be as successful as we want it to be, Search also needs to be robust.